lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -317,6 +323,12 @@ def check_FunctionDefinition(o)
     #  but not be special in any other way
     #
     check_NamedDefinition(o)
+
+    # This is perhaps not ideal but it's very difficult to pass a ReservedWord through
+    # the mechanism that creates qualified names (namestack, namepop etc.)
+    if FUTURE_RESERVED_WORDS[o.name]
+      acceptor.accept(Issues::FUTURE_RESERVED_WORD, o, {:word => o.name})
```

Comment:
@thallgren @hlindberg can you check this spot in particular? The structure of check_FunctionDefinition changed a bit, so I'm not sure whether this is right or if the fix for [PUP-5036](https://tickets.puppetlabs.com/browse/PUP-5036) even needs to be merged up.